src/core_plugins/kibana/public/dashboard/dashboard_container_api.js

```diff
@@ -0,0 +1,38 @@
+import { ContainerAPI } from 'ui/embeddable';
+
+export class DashboardContainerAPI extends ContainerAPI {
+  constructor(dashboardState, queryManager) {
+    super();
+    this.dashboardState = dashboardState;
+    this.queryManager = queryManager;
+  }
+
+  addFilter(field, value, operator, index) {
+    this.queryManager.add(field, value, operator, index);
+  }
+
+  updatePanel(panelIndex, panelAttributes) {
+    const panelToUpdate = this.dashboardState.getPanels().find((panel) => panel.panelIndex === panelIndex);
+    Object.assign(panelToUpdate, panelAttributes);
+    this.dashboardState.saveState();
+    return panelToUpdate;
+  }
+
+  getAppState() {
+    return this.dashboardState.appState;
+  }
+
+  getIsViewOnlyMode() {
+    return this.dashboardState.getIsViewMode();
+  }
+
+  getInitialState(path, uiState) {
```

Comment:
hah, okay, yea, this function is carried over from the past and I never understood much of it.  Will switch to your intended version. :)